src/com/sap/piper/integration/TransportManagementService.groovy

```diff
@@ -51,9 +51,14 @@ class TransportManagementService implements Serializable {
         }
 
         def response = sendApiRequest(parameters)
+
+        if (config.verbose) {
+            echo("Received response with status ${response.status} from authentification request.")
+        }
+
         echo("OAuth Token retrieved successfully.")
```

Comment:
The code `echo("OAuth Token retrieved successfully.")` was the same before and also the message was bogus already before since it might be wrong depending on the http response code.

Focus of this PR (see title) is to avoid having the token exposed in the log. The code you are claiming about was the same before. Of course it is reasonable to apply further improvements, but I would like to suggest to do this in another PR.  --> Would you like to provide another PR for that, maybe after merging this one?

With this PR I moved the response out of `sendApiRequest`. Before the response was only availabe inside. It was not possible to be more precise before ...

By the way: is there a documentation available explaining the response codes. For getting the authToken it would also be possible to return 201 (created).

> method can be shorten to the very expected code

I think the way httpPlugin deals with http response codes in bogus. When we restrict the response code to 200 we will get an exception in all the other cases. In that cases it is undefined what happens with the response (maybe the response body is part of the exception message, maybe not, there is no contract). The big problem behind is basically that reponses like eg 4xx, 5xx are handled as exceptions. Getting a 4xx, 5xx is completely normal from the http communication point of view and in this case the response body might contain some hints, especially with http-400 (bad request) where it might be explained what was wrong with the request.

From that point of view it might be better to accept also 4xx and 5xx responses and check the response code ourselfs. In that case we will have the `response.content` available in order to provide more details.

Reply:
> Would you like to provide another PR for that, maybe after merging this one?

Yes, will do so.